src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -65,6 +74,53 @@ public List<ExtensionRestHandler> getExtensionRestHandlers() {
             );
     }
 
+    @Override
+    public Collection<Object> createComponents(ExtensionsRunner runner) {
+
+        SDKRestClient sdkRestClient = getRestClient();
+        SDKClusterService sdkClusterService = runner.getSdkClusterService();
+        Settings environmentSettings = runner.getEnvironmentSettings();
```

Comment:
For now this is ok. Note to self we probably need a settings wrapper object similar to how we're wrapping the NamedXContentRegistry for exactly the same reasons.